bpf/lib/nodeport.h

```diff
@@ -695,6 +696,19 @@ int tail_nodeport_ipv6_dsr(struct __ctx_buff *ctx)
 #elif DSR_ENCAP_MODE == DSR_ENCAP_NONE
 	ret = dsr_set_ext6(ctx, ip6, &addr, port, &ohead);
 #elif DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+
+#if defined(ENABLE_WIREGUARD)
```

Comment:
I think we don't need to be WG specific. We could check whether output iface requires adding L2 - see https://github.com/cilium/cilium/pull/33421.

Reply:
I was trying to avoid compiling in this code when not needed as ci-verifier was complaining about a complexity issue